Sources/ContextPropagation/Context.swift

```diff
@@ -0,0 +1,29 @@
+public protocol ContextKey {
+    associatedtype Value
+}
+
+public struct Context {
+    private var dict = [ObjectIdentifier: ValueContainer]()
+
+    public init() {}
+
+    public mutating func inject<Key: ContextKey>(_ key: Key.Type, value: Key.Value) {
```

Comment:
I expect those will change a lot, but let's get there when we get there -- driven by use cases, which we should start writing up next (as tests or mini apps perhaps) to inform the design :-)

Kind of hopping ahead here, but as we talked in our discussion about "where" the logic to inject/extract has to live, so that will inform how these APIs have to look like. If we'll end up with some "Injector" or similar protocols that can "inject into an HTTPRequest" or similar etc.

Reply:
Yes, I think we can tackle the "injector" once we have multiple use cases in place to figure out where to best build abstractions.

Reply:
Ok :)